cluster-autoscaler/metrics/metrics.go

```diff
@@ -536,6 +554,24 @@ func UpdateNodeGroupTargetSize(targetSizes map[string]int) {
 	}
 }
 
+// UpdateNodeGroupHealthStatus records if node group is healthy to autoscaling
+func UpdateNodeGroupHealthStatus(nodeGroup string, healthy bool) {
+	if healthy {
+		nodesGroupHealthiness.WithLabelValues(nodeGroup).Set(1)
+	} else {
+		nodesGroupHealthiness.WithLabelValues(nodeGroup).Set(0)
+	}
+}
+
+// UpdateNodeGroupBackOffStatus records if node group is backoff for not autoscaling
+func UpdateNodeGroupBackOffStatus(nodeGroup string, backOff bool, reason string) {
+	if backOff {
+		nodeGroupBackOffStatus.WithLabelValues(nodeGroup, reason).Set(1)
```

Comment:
This will not work the way you want: once the backoff is over, the value of 1 will keep on being emitted for the old reason. Consider:

1. `UpdateNodeGroupBackOffStatus("foo", true, "reason")`

   Under /metrics endpoint you'll get something like:
   
   ```
   node_group_backoff_status{node_group=foo,reason=reason} 1
   ```

2. `UpdateNodeGroupBackOffStatus("foo", false, "")`

   Under /metrics endpoint you'll get something like:
   
   ```
   node_group_backoff_status{node_group=foo,reason=reason} 1
   node_group_backoff_status{node_group=foo,reason=""} 0
   ```

What you may want to do instead is to use a string metric with value equal to the backoff reason.

Reply:
I agree with the suggestion and will make the necessary modifications as soon as possible.

Reply:
Modifications done and PR submitted.

Reply:
Apologies for not being clear - I think the current version of the code will have exactly the same issue: I expect `reason` to be an empty string every time `backoff` is `false`. If you want to track backoff reasons like this, you need to keep track of the last `reason` value and explicitly clear it when `backoff` becomes `false`. Alternatively, you could implement a custom collector that will only emit metrics with up-to-date reason. This would be the cleanest solution. With the existing code, in the scenario from my previous comment, after backoff is over, you get:

```
node_group_backoff_status{node_group=foo,reason=reason} 1
node_group_backoff_status{node_group=foo,reason=""} 0
```

If you instead track nodegroup -> reason mapping and clear the last reason you will instead get:

```
node_group_backoff_status{node_group=foo,reason=reason} 0
```

If you decide to write a custom collector, once backoff is over, you could simply get empty output:

```
```

Reply:
Thank you for raising this issue, and you’re right, I misunderstood it. It is indeed not expected behavior that the previous nodeGroup backoff metric continues to be reported as 1 after the backoff is over. I will take some time to reconsider how to address this and submit the code changes as soon as possible.

Reply:
Modifications done and PR submitted.

Reply:
@x13n  Please review my changes